File: src/main/java/com/arangodb/intellij/aql/util/DataSourceWindowCallback.java
package com.arangodb.intellij.aql.util;


import com.arangodb.intellij.aql.actions.AqlDataService;
import com.arangodb.intellij.aql.toolWindow.ServerToolWindow;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowManager;

public class DataSourceWindowCallback implements ActionMessageCallback {
    private final Project project;

    public DataSourceWindowCallback(final Project project) {
        this.project = project;
    }

    @Override
    public void call() {
        final ToolWindow window = ToolWindowManager.getInstance(project).getToolWindow(ServerToolWindow.WINDOW_ID);
        if (window != null) {
            window.activate(null, true);
        }
        AqlDataService.with(project).showServerDialog();
    }
}

File: src/main/kotlin/com/arangodb/intellij/aql/util/DataSourceWindowCallback.kt
package com.arangodb.intellij.aql.util

import com.arangodb.intellij.aql.actions.AqlDataService
import com.arangodb.intellij.aql.toolWindow.ServerToolWindow
import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindowManager

class DataSourceWindowCallback(private val project: Project) : ActionMessageCallback {
    override fun call() {
        ToolWindowManager.getInstance(project).getToolWindow(ServerToolWindow.WINDOW_ID)?.activate(null, true)
        AqlDataService.with(project).showServerDialog()
    }
}
